changelogs/unreleased/6691-catch-all-arguments-fix.yml

```diff
@@ -0,0 +1,5 @@
+change-type: patch
+description: Fix bugs introduced in https://github.com/inmanta/inmanta-core/pull/6808
+destination-branches: [master]
+sections:
+  minor-improvement: "Stricter type checking of values returned by plugins.  Specifying None as returned type requires None to be returned."
```

Comment:
Once this is merged in master, I will create another PR for iso6, and:
1. Merge the fix in this PR
2. Allow any returned value for plugins that use None as returned value annotation